src/ai/backend/manager/repositories/container_registry/repository.py

```diff
@@ -37,6 +54,113 @@ class ContainerRegistryRepository:
     def __init__(self, db: ExtendedAsyncSAEngine) -> None:
         self._db = db
 
+    @container_registry_repository_resilience.apply()
+    async def create_registry(
+        self,
+        creator: ContainerRegistryCreator,
+    ) -> ContainerRegistryData:
+        async with self._db.begin_session() as db_session:
+            reg_row = ContainerRegistryRow.from_creator(creator)
+            db_session.add(reg_row)
+            await db_session.flush()
+            await db_session.refresh(reg_row)
+
+            if creator.allowed_groups:
+                await self._update_registry_access_allowed_groups(
+                    db_session, reg_row.id, creator.allowed_groups
+                )
+
+            return reg_row.to_dataclass()
```

Comment:
Let's apply Creator pattern here using `ai.backend.manager.repositories.base.creator.Creator` and `ai.backend.manager.repositories.base.creator.execute_creator()`